ompi/mca/coll/hcoll/coll_hcoll_module.c

```diff
@@ -290,7 +280,8 @@ mca_coll_hcoll_comm_query(struct ompi_communicator_t *comm, int *priority)
            mxm bcol in libhcoll needs world_group fully functional during init
            world_group, i.e. ompi_comm_world, is not ready at hcoll component open
            call */
-        opal_progress_register(mca_coll_hcoll_progress);
+        printf("hcoll_progress_fn=%p\n", hcoll_progress_fn);
+        opal_progress_register(hcoll_progress_fn);
```

Comment:
The code will be more readable if the progress registration was the last step of a successful initialization.

Reply:
In fact the progress function should only be registered when coll_module_enable is called.

Reply:
@jladd-mlnx @vspetrov what do you think? is it possible to move or is progress needed for some phase during initialization?

Reply:
@vspetrov do we need it early for non-blocking context create?

Reply:
No, i don't think we need hcoll internal progress to be registered into opal before we launch first collective via hcoll.